saleor/graphql/discount/mutations/promotion_update.py

```diff
@@ -0,0 +1,57 @@
+import graphene
+from django.core.exceptions import ValidationError
+
+from ....discount import models
+from ....permission.enums import DiscountPermissions
+from ...core import ResolveInfo
+from ...core.descriptions import ADDED_IN_315, PREVIEW_FEATURE
+from ...core.doc_category import DOC_CATEGORY_DISCOUNTS
+from ...core.mutations import ModelMutation
+from ...core.types import Error
+from ...core.validators import validate_end_is_after_start
+from ..enums import PromotionUpdateErrorCode
+from ..types import Promotion
+from .promotion_create import PromotionInput
+
+
+class PromotionUpdateError(Error):
+    code = PromotionUpdateErrorCode(description="The error code.", required=True)
+
+
+class PromotionUpdateInput(PromotionInput):
+    name = graphene.String(description="Promotion name.")
+
+
+class PromotionUpdate(ModelMutation):
+    class Arguments:
+        id = graphene.ID(required=True, description="ID of the promotion to update.")
+        input = PromotionUpdateInput(
+            description="Fields required to update a promotion.", required=True
+        )
```

Comment:
Is it intended to not be able to update `rules`?

Reply:
Yes. There will be separate mutations for updating the rules